src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,9 +32,24 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
-  private readTasks(): Task[] {
+  private readTasks(query?:string): Task[] {
+
     const tasks = localStorage.getItem(LocalTaskService.STORAGE_KEY);
-    return tasks ? JSON.parse(tasks) : [];
+    const tasksList = JSON.parse(tasks);
+    const searchResult = [];
+
+    if (tasks && !query) {
+      return tasksList;
+    } else if (query) {
+      tasksList.forEach((task)=> {
```

Comment:
you can do this in a one-liner by filtering on tasksList and get rid of the` const searchResult` altogether